src/map/entities/charentity.h

```diff
@@ -437,6 +438,20 @@ class CCharEntity : public CBattleEntity
     auto getPacketListCopy() -> std::deque<std::unique_ptr<CBasicPacket>>; // Return a COPY of packet list
     void clearPacketList();
 
+    template <typename T>
+    void queuePacket(T pkt)
```

Comment:
This could do with a comment explaining why it exists and how it's different to pushPacket - saying it builds the header and packet based on the decomp definitions and then forwards it to pushPacket is fine.